### PR TITLE
lmfit 5.1 (new formula)

### PR DESCRIPTION
**Pull request by T045T**
Added a formula for lmfit ( http://apps.jcns.fz-juelich.de/doku/sc/lmfit ).

No test block because I wasn't sure about compiling a test program in a test - both how to best do it and whether it should be done at all.

### CODE REVIEW

File: lmfit.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  def install
+    system "./configure", "--disable-dependency-tracking",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.c").write <<-EOS
+    #include "lmcurve.h"
+    #include "lmmin.h"
```
Comment (tdsmith) [inline lmfit.rb]: Can you indent the test code another 2 spaces?

**Comment by tdsmith** [comment]
The test looks good! If you want you can install the demo from the tarball with e.g. `(share/"lmfit").install Dir["demo/*"]` or whatever and then build it from there in the test block.

**Comment by dpo** [comment]
+1 for the demo.

**Comment by T045T** [comment]
I changed the formula to install the demo sources - since they're supposed to demonstrate how the library works, I thought this made more sense than copying the binaries.

I also added a caveat to inform the user about those demo files. Is that okay?

**Comment by dpo** [comment]
Great. Thank you for your contribution!

**Pull request closed without merging**